extensions/kafka-connect-jdbc/src/main/java/io/confluent/connect/jdbc/sink/BufferedRecords.java

```diff
@@ -178,13 +178,38 @@ public List<SinkRecord> add(SinkRecord record) throws SQLException, TableAlterOr
       // no value schema, value is a map, must convert to Struct
       if (!(record.value() instanceof HashMap)) {
         log.warn("auto schema convertToStruct only support hashmap to struct");
+        throw new SQLException("auto schema convertToStruct only support hashmap to struct");
       }
-      Object structValue = convertToStruct(valueSchema, record.value());
-      record = new SinkRecord(record.topic(), record.kafkaPartition(),
-          record.keySchema(), record.key(),
-          valueSchema, structValue,
-          record.kafkaOffset(), record.timestamp(), record.timestampType(), record.headers());
+      HashMap<String, Object> valueMap = (HashMap) record.value();
+      List<HashMap<String, Object>> list = (List<HashMap<String, Object>> )valueMap.get("data");
+      boolean isDelete = false;
+      if (((String)valueMap.get("type")).toLowerCase().equals("delete")) {
+        isDelete = true;
```

Comment:
Only check delete? So we should set insert mode to the update mode to do update/delete?

Reply:
we can create latest 1 table to delete old data automatically.